### PR TITLE
Clock from Miri

### PR DESCRIPTION
**Pull request by miribar**
Hi Ameed,

I've added Clock & ClockBonus1.
Hope I found all the bugs.. 😉 

Thanks!

### CODE REVIEW

**Comment by miribar** [comment]
Hi Ameed,

I've added the ClockBonus1, hope I got all the bugs 😉 
Shabbat Shalom!

File: src/ameedf/jb8223/assignments/hw02/Clock.java [hunk 1]
```diff
@@ -34,4 +34,106 @@
...
  */
 public class Clock {
+
+    private int hours;
+    private int minutes;
+    private int seconds;
+    private int milliseconds;
+
+    private final int minValueAllowed = 0;
+    private final int maxValueAllowed = 1_000_000_000;
```
Comment (ameedf) [inline src/ameedf/jb8223/assignments/hw02/Clock.java:44]: if you want to define the variable as final, then you should also define it as **static**.. but we'll learn about this soon

File: src/ameedf/jb8223/assignments/hw02/Clock.java [hunk 2]
```diff
@@ -34,4 +34,106 @@
...
+    public void addHours(int hoursToAdd) {
+        if (hoursToAdd >= minValueAllowed && hoursToAdd <= maxValueAllowed) {
+                this.hours = (this.hours + hoursToAdd) % 24;
+        }
+    }
+
+    // addMinutes - with a single parameter which represents the number of minutes to add.
+    public void addMinutes(int minutesToAdd) {
+        if (minutesToAdd >= minValueAllowed && minutesToAdd <= maxValueAllowed) {
+                addHours(minutesToAdd / 60);
```
Comment (ameedf) [inline src/ameedf/jb8223/assignments/hw02/Clock.java:72]: BUG: what happens if the clock is 10:59:12.013 and you want to add 2 minutes ? does the hours variable change?

File: src/ameedf/jb8223/assignments/hw02/Clock.java [hunk 3]
```diff
@@ -34,4 +34,106 @@
...
+                this.minutes = (this.minutes + minutesToAdd) % 60;
+        }
+    }
+
+    // addSeconds - with a single parameter which represents the number of seconds to add.
+    public void addSeconds(int secondsToAdd) {
+        if (secondsToAdd >= minValueAllowed && secondsToAdd <= maxValueAllowed) {
+                addMinutes(secondsToAdd / 60);
```
Comment (ameedf) [inline src/ameedf/jb8223/assignments/hw02/Clock.java:80]: same bug as above

File: src/ameedf/jb8223/assignments/hw02/Clock.java [hunk 4]
```diff
@@ -34,4 +34,106 @@
...
+        else
+            System.out.printf(seconds + ".");
+
+        if (milliseconds < 10)
+            System.out.printf("00" + milliseconds + " ");
+        else if (milliseconds < 100)
+            System.out.printf("0" + milliseconds + " ");
+        else
+            System.out.printf(milliseconds + " ");
+    }
```
Comment (ameedf) [inline src/ameedf/jb8223/assignments/hw02/Clock.java:138]: 1. try to use " { } " in if-else statements
2. it's better to build up one string and use the System.out.print/ln once

File: src/ameedf/jb8223/assignments/hw02/Clock.java [hunk 5]
```diff
@@ -34,4 +34,106 @@
...
+                System.out.println("AM");
+            }
+            hours = hoursValueKeeper;
+        }
+    }
+
+    // addZeroOrNotAndPrint - a helper method that does the actual time fraction printing
+    private void addZeroOrNotAndPrint() {
+        if (hours < 10)
+            System.out.printf("0" + hours + ":");
```
Comment (ameedf) [inline src/ameedf/jb8223/assignments/hw02/Clock.java:118]: printf is used to format strings - we'll learn about that soon.
Here, it's better to use "print" and not "printf"

**Review by ameedf — COMMENTED** [review]
very good job

File: src/ameedf/jb8223/assignments/hw02/Clock.java (hunk 3 above)
Comment (miribar) [reply src/ameedf/jb8223/assignments/hw02/Clock.java:80]: You're right, i'll fix it.

File: src/ameedf/jb8223/assignments/hw02/Clock.java (hunk 4 above)
Comment (miribar) [reply src/ameedf/jb8223/assignments/hw02/Clock.java:138]: Wanted to avoid one big, messy string.

**Comment by miribar** [comment]
Thank you :-)

File: src/ameedf/jb8223/assignments/hw02/Clock.java (hunk 4 above)
Comment (ameedf) [reply src/ameedf/jb8223/assignments/hw02/Clock.java:138]: you didn't.. actually it's out there - between each 2 "System.out.print(..."
:)

Comment (miribar) [reply src/ameedf/jb8223/assignments/hw02/Clock.java:138]: got your point, will revise.

**Pull request closed without merging**